src/libraries/System.Private.CoreLib/src/System/Buffers/Utilities.cs

```diff
@@ -29,5 +29,41 @@ internal static int GetMaxSizeForBucket(int binIndex)
             Debug.Assert(maxSize >= 0);
             return maxSize;
         }
+
+        internal enum MemoryPressure
+        {
+            Low,
+            Medium,
+            High
+        }
+
+        internal static MemoryPressure GetMemoryPressure()
+        {
+            const double HighPressureThreshold = .90;       // Percent of GC memory pressure threshold we consider "high"
+            const double MediumPressureThreshold = .70;     // Percent of GC memory pressure threshold we consider "medium"
+
+            GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+            if (memoryInfo.MemoryLoadBytes >= memoryInfo.HighMemoryLoadThresholdBytes * HighPressureThreshold)
+            {
+                return MemoryPressure.High;
+            }
+
+            if (memoryInfo.MemoryLoadBytes >= memoryInfo.HighMemoryLoadThresholdBytes * MediumPressureThreshold)
+            {
+                return MemoryPressure.Medium;
+            }
+
+            return MemoryPressure.Low;
+        }
+
+        internal static bool TrimBuffers { get; } =
```

Comment:
Do we still need this switch? I do not think it was ever used by anybody.

Reply:
Let's remove it.